### PR TITLE
Demo fixing

### PR DESCRIPTION
**Pull request by JulianoGianlupi**
Fixed 7 demos that failed to run. Moved 10 demos that failed to run to the experimental folder (as they require functionalities that are not yet present in CompuCell3D).

### Fixed demos:

Demos/Models/FitzHughNagumo_OpenCL/
Demos/SteppableDemos/DiffusionSolverFE_OpenCL/DiffusionSolverFE_OpenCL/
Demos/SteppableDemos/DiffusionSolverFE_OpenCL/DiffusionSolverFE_OpenCL_3D
Demos/SteppableDemos/ReactionDiffusionSolverFE_OpenCL_Implicit/ReactionDiffusion_OpenCL_Implicit_3D/
Demos/SteppableDemos/ReactionDiffusionSolverFE_OpenCL_Implicit/SimpleDiffusion/
Demos/SteppableDemos/DiffusionSolverFE/diffusion_3D_scale_wall/
Demos/optimization/optimization_demo/

### Moved demos

Demos/PluginDemos/AdhesionFlexPython/
Demos/SteppableDemos/FlexibleDiffusionSolverFE/Steering/
Demos/PluginDemos/elasticityTest_steering/
Demos/PluginDemos/elongationTest_steering/
Demos/CompuCellPythonTutorial/vectorPlot/
Demos/CompuCellPythonTutorial/clusterMitosisPeriodicBC/
Demos/CompuCellPythonTutorial/FiPyConcentrationTest/
Demos/experimental/FiPyConcentrationTest/
Demos/CompuCellPythonTutorial/GraphVTKVis/
Demos/experimental/GraphVTKVis/

### CODE REVIEW

**Comment by JulianoGianlupi** [comment]
Also, Jim Sluka removed outdated parameters from XMLs and performed fixes.

File: CompuCell3D/core/Demos/CompuCellPythonTutorial/clusterMitosisPeriodicBC/Simulation/clusterMitosisSteppables.py [hunk 1]
```diff
@@ -5,64 +5,65 @@
 from random import uniform
```
[the remark this answers is not in the document]
Comment (JulianoGianlupi) [reply CompuCell3D/core/Demos/CompuCellPythonTutorial/clusterMitosisPeriodicBC/Simulation/clusterMitosisSteppables.py]: But in this one SteppableBasePy is being used

**Comment by jpsluka** [comment]
"I have an impression that this PR is based on a very old snapshot of master." The Demo's we tested are the ones that were being distributed with CC3D 3.7.7. If the demo set in that distro is different than what is in the repository then there is some flaw in how distros are being built.

The cases where the old steppable base (SteppablePy) is used is, again, what was being distributed with CC3D.

**Comment by CompuCell3D** [comment]
Jim the latest distribution uses SteppableBasePy as a base class. There are literally 6 exceptions where on purpose we left SteppablePy as a base class and those steppables are not actually instantiated. The code is there to guide people how to transition away from SteppablePy.

File: CompuCell3D/core/Demos/optimization/optimization_demo/Simulation/optimization_demo.xml [hunk 2]
```diff
@@ -6,7 +6,7 @@
       <Dimensions x="100" y="100" z="1"/>
       <Steps>200</Steps>
       <RandomSeed>20</RandomSeed>
-      <Temperature>{{TEMPERATURE}}</Temperature>
+      <Temperature>10</Temperature>
```
[the remark this answers is not in the document]
Comment (JulianoGianlupi) [reply CompuCell3D/core/Demos/optimization/optimization_demo/Simulation/optimization_demo.xml]: Ok, but leaving {{TEMPERATURE}} and {{CONTACT_A_B}} causes an error. I've added an error.txt file to \Demos\optimization\optimization_demo, I'll commit it soon.

`Traceback (most recent call last):
  File "C:\CompuCell3D-64bit\pythonSetupScripts\CompuCellPythonSimulationNewPlayer.py", line 351, in <module>
    execfile(CompuCellSetup.simulationPaths.simulationPythonScriptName)
  File "C:\Users\jferrari\Documents\CompuCell3D - dev stuff\code\CompuCell3D\core\Demos\optimization\optimization_demo\Simulation\optimization_demo.py", line 27, in <module>
    CompuCellSetup.mainLoop(sim,simthread,steppableRegistry)
  File "C:\CompuCell3D-64bit\pythonSetupScripts\CompuCellSetup.py", line 2262, in mainLoop
    return mainLoopNewPlayer(sim, simthread, steppableRegistry, _screenUpdateFrequency)
  File "C:\CompuCell3D-64bit\pythonSetupScripts\CompuCellSetup.py", line 1826, in mainLoopNewPlayer
    steppableRegistry.finish()
  File "C:\CompuCell3D-64bit\pythonSetupScripts\PySteppables.py", line 1656, in finish
    steppable.finish()
  File "C:\Users\jferrari\Documents\CompuCell3D - dev stuff\code\CompuCell3D\core\Demos\optimization\optimization_demo\Simulation\optimization_demoSteppables.py", line 48, in finish
    temp = float(self.getXMLElementValue(*temperature_access_path))
ValueError: could not convert string to float: {{TEMPERATURE}}`

**Comment by JulianoGianlupi** [comment]
I've done the requested changes, Maciek.

If anything else needs to be done just ask.

**Comment by jpsluka** [comment]
Perhaps we should move those demos to a folder like "old and doesn't work anymore dont bother testing" :)
It is fine to have old examples, but if we do, we can't run tests on new distributions without constantly having those old variants failing.

Perhaps we will have the same problem with the old Potts "temperature" versus the new "FluctuationAmplitude". Most of the demos still use "temperature". I suspect that "temperature" is still used in a few places in the documentation. Tweditt++ (3.7.9 distro) still uses "temperature" for newly created models.

File: CompuCell3D/core/Demos/CompuCellPythonTutorial/clusterMitosis/Simulation/clusterMitosisSteppables.py [hunk 3]
```diff
@@ -42,7 +42,10 @@ def step(self,mcs):
             
             
             if clusterVolume>250: # condition under which cluster mitosis takes place
-                mitosisClusterIdList.append(clusterId) # instead of doing mitosis right away we store ids for clusters which should be divide. This avoids modifying cluster list while we iterate through it
+                # instead of doing mitosis right away we store ids for clusters 
+				#which should be divide. This avoids modifying cluster list
```
Comment (CompuCell3D) [inline CompuCell3D/core/Demos/CompuCellPythonTutorial/clusterMitosis/Simulation/clusterMitosisSteppables.py]: let's keep indent level of 4 spaces

File: CompuCell3D/core/Demos/SteppableDemos/DiffusionSolverFE/diffusion_2D_scale_wall/Simulation/diffusion_2D_scale_wall.xml [hunk 4]
```diff
@@ -16,6 +14,7 @@
     <CellType TypeName="Medium" TypeId="0"/>
     <CellType TypeName="Amoeba" TypeId="1"/>
     <CellType TypeName="Bacteria" TypeId="2"/>
+<!--     <CellType TypeName="Wall" TypeId="2" Freeze=""/> -->
```
Comment (CompuCell3D) [inline CompuCell3D/core/Demos/SteppableDemos/DiffusionSolverFE/diffusion_2D_scale_wall/Simulation/diffusion_2D_scale_wall.xml]: remove this line

File: CompuCell3D/core/Demos/experimental/CompuCellPythonTutorial/vectorPlot/error.txt [hunk 5]
```diff
@@ -0,0 +1,12 @@
+Traceback (most recent call last):
```
Comment (CompuCell3D) [inline CompuCell3D/core/Demos/experimental/CompuCellPythonTutorial/vectorPlot/error.txt]: this error is caused by the fact that you are using SteppablePy as a base class. Let's fix if

File: CompuCell3D/core/Demos/experimental/PluginDemos/AdhesionFlexPython/Simulation/AdhesionFlexSteppables.py [hunk 6]
```diff
@@ -6,19 +6,34 @@
 
 class AdhesionMoleculesSteppables(SteppableBasePy):
     def __init__(self,_simulator,_frequency=1):
-        SteppableBasePy.__init__(self,_simulator, _frequency)
+        SteppablePy.__init__(self,_frequency)
+        self.simulator=_simulator
```
Comment (CompuCell3D) [inline CompuCell3D/core/Demos/experimental/PluginDemos/AdhesionFlexPython/Simulation/AdhesionFlexSteppables.py]: lines 10-14 are not needed with SteppableBasePy

Comment (CompuCell3D) [reply CompuCell3D/core/Demos/experimental/PluginDemos/AdhesionFlexPython/Simulation/AdhesionFlexSteppables.py]: I would not touch this demo i.e use the demo from current master

File: CompuCell3D/core/Demos/experimental/PluginDemos/AdhesionFlexPython/error.txt [hunk 7]
```diff
@@ -0,0 +1,16 @@
+Traceback (most recent call last):
```
Comment (CompuCell3D) [inline CompuCell3D/core/Demos/experimental/PluginDemos/AdhesionFlexPython/error.txt]: again this problem is related to wrong base class for the steppable. let's fix it by reverting to the demo version from current master branch

File: CompuCell3D/core/Demos/experimental/PluginDemos/elasticityTest_steering/Simulation/elasticityTest_steering_steppables.py [hunk 8]
```diff
@@ -6,8 +6,8 @@
 
 class ElasticitySteering(SteppableBasePy):
     def __init__(self,_simulator,_frequency=1):
-        SteppableBasePy.__init__(self,_simulator, _frequency)
-        
+        SteppablePy.__init__(self,_frequency)
```
Comment (CompuCell3D) [inline CompuCell3D/core/Demos/experimental/PluginDemos/elasticityTest_steering/Simulation/elasticityTest_steering_steppables.py]: Again, use version of the demo from the master branch -

Comment (CompuCell3D) [inline CompuCell3D/core/Demos/experimental/PluginDemos/elasticityTest_steering/Simulation/elasticityTest_steering_steppables.py]: you have SteppableBasePy class but you are calling constructor for SteppablePy

File: CompuCell3D/core/Demos/experimental/PluginDemos/elasticityTest_steering/Simulation/elasticityTest_steering_steppables.py [hunk 9]
```diff
@@ -6,8 +6,8 @@
...
+        self.simulator=_simulator
```
Comment (CompuCell3D) [inline CompuCell3D/core/Demos/experimental/PluginDemos/elasticityTest_steering/Simulation/elasticityTest_steering_steppables.py]: not needed

File: CompuCell3D/core/Demos/experimental/PluginDemos/elongationTest_steering/Simulation/elongationTest_steering_steppables.py [hunk 10]
```diff
@@ -8,8 +8,8 @@
             
 class LengthConstraintSteering(SteppableBasePy):
     def __init__(self,_simulator,_frequency=100):
-        SteppableBasePy.__init__(self,_simulator,_frequency)
-        
+        SteppablePy.__init__(self,_frequency)
```
Comment (CompuCell3D) [inline CompuCell3D/core/Demos/experimental/PluginDemos/elongationTest_steering/Simulation/elongationTest_steering_steppables.py]: again use version of the demo from the current master branch. It works.

File: CompuCell3D/core/Demos/experimental/SteppableDemos/FlexibleDiffusionSolverFE/Steering/Simulation/diffusion_2D_steering_steppables.py [hunk 11]
```diff
@@ -6,9 +6,10 @@
 
 
 
-class DiffusionSolverSteering(SteppableBasePy):
+class DiffusionSolverSteering(SteppablePy):
```
Comment (CompuCell3D) [inline CompuCell3D/core/Demos/experimental/SteppableDemos/FlexibleDiffusionSolverFE/Steering/Simulation/diffusion_2D_steering_steppables.py]: use version of the demo from the current master branch. It works.

**Comment by CompuCell3D** [comment]
We are moving to 4.0.0 CC3D and those fixes no longer apply

**Pull request closed without merging**